In [2]:
import numpy as np
import pandas as pd
import re
 
#a = df['COST (£) charged to Wellcome (inc VAT when charged)'] = (df['COST (£) charged to Wellcome (inc VAT when charged)'].apply(lambda x: ''.join(list(filter(str.isdigit, str(x))))))


In [46]:
df = pd.read_csv('C:/Users/keith/Desktop/Thinkful/WELLCOME_APCspend2013_forThinkful.csv',encoding = "ISO-8859-1")
pd.set_option('display.max_columns', None) # disables trucated columns 
pd.set_option('display.max_rows', None) # disables trucated rows
df.head()
df = df.sort_values('Journal title')
df[['Journal title' ]].groupby('Journal title').head()

Journal title
21                                 ACS Chemical Biology
20                                 ACS Chemical Biology
8                                  ACS Chemical Biology
9                                  ACS Chemical Biology
19                                 ACS Chemical Biology
22                            ACS Chemical Neuroscience
23                                             ACS NANO
34                                             ACS Nano
924                                              ACTA F
1808                                                AGE
2125                                               AIDS
2123                                               AIDS
2115                                               AIDS
1797                                         AIDS Behav
1811                                          AIDS Care
1824                                          AIDS Care
2116                                       AIDS Journal
235                           AIDS Research and Therapy
2117                                            AIDS UK
2118                                            AIDS UK
1317                                          ASN Neuro
439                  Academy of Nutrition and Dietetics
927   Acta Crystallographica Section D,  Biological ...
920   Acta Crystallographica Section D: Biological C...
929   Acta Crystallographica Section F: Structural B...
928   Acta Crystallographica Section F: Structural B...
921                   Acta Crystallographica, Section D
922                              Acta Crystallography D
923                                              Acta D
1711                         Acta Dermato Venereologica
1716                                 Acta Diabetologica
1717                                   Acta Neuropathol
1718                              Acta Neuropathologica
1719                              Acta Neuropathologica
1720                              Acta Neuropathologica
2051                                Acta Opthalmologica
1871                                       Acta Physiol
1873                                          Addiction
1872                                          Addiction
1721      Advances in Experimental Medicine and Biology
1723      Advances in Experimental Medicine and Biology
1724      Advances in Experimental Medicine and Biology
1722      Advances in Experimental Medicine and Biology
1803                                                Age
1186                                     Age and Ageing
353                                    Ageing & Society
1874                                         Aging Cell
1813                                          Aids Care
1812                                          Aids Care
1109                             Alcohol and Alcoholism
1875            Alimentrary Pharmacology & Therapeutics
1876            Alimentrary Pharmacology & Therapeutics
1822                                        Am J Bioeth
1868                                  Am J Trop Med Hyg
44                       Americal Journal of Psychiatry
24                            American Chemical Society
1877                               American Ethnologist
2007                               American Ethnologist
1110                          American Jnl Epidemiology
416             American Journal for Clinical Nutrition
1825        American Journal of Bioethics--Neuroscience
417              American Journal of Clinical Nutrition
1187                   American Journal of Epidemiology
1112                   American Journal of Epidemiology
1111                   American Journal of Epidemiology
440            American Journal of Geriatric Psychiatry
374                  American Journal of Human Genetics
441                  American Journal of Human Genetics
442                  American Journal of Human Genetics
443                  American Journal of Human Genetics
444                  American Journal of Human Genetics
934       

In [4]:
journals = df['Journal title']

df['Journal title'] = journals.str.replace(' ', '')
df['Journal title'] = journals.str.replace('.', '')
df['Journal title'] = journals.apply(lambda x : re.sub('([Pp][Ll][Oo][Ss][Oo][Nn][Ee]|([Pp][Ll][Oo][Ss][1]))',"Plos One", str(x)))
df['Journal title'] = journals.apply(lambda x : re.sub('(\w*Biological\w*)',"Journal of Biological Chemistry", str(x)))
df['Journal title'] = journals.apply(lambda x : re.sub('(\w*[Nn]ucleic[Aa]c\w*)',"Nucleic Acids Research", str(x)))
df['Journal title'] = journals.apply(lambda x : re.sub('(\w*(nalAc\w*)|(PNAS)\w*)',"Proceedings of the National Academy of Sciences", str(x)))
df['Journal title'] = journals.apply(lambda x : re.sub('(\w*nMol\w*)',"Human Molecular Genetics", str(x))) 
df['Journal title'] = journals.apply(lambda x : re.sub('(\w*([Nn]euro[Iil]mage)\w*)',"Neuro Image", str(x))) 
 

In [5]:
journals.value_counts().head(5)

Plos One                                           207
Journal of Biological Chemistry                     64
Proceedings of the National Academy of Sciences     37
Neuro Image                                         31
Nucleic Acids Research                              26
Name: Journal title, dtype: int64

### The five most common journals.
Plos One count = 207<br>
Journal of Biological Chemistry count = 64<br>
Proceedings of the National Academy of Sciences count = 37<br>
Neuro Image = count 31<br>
Nucleic Acids Research count = 26<br>

In [6]:
cost = df['COST (£) charged to Wellcome (inc VAT when charged)']
cost = cost.str.replace('£', '')
cost = cost.str.replace('$','')
cost = cost.astype(float)
df['COST (£) charged to Wellcome (inc VAT when charged)'] = cost

In [7]:
a = df[df['Journal title']=='Plos One'] 
Plos_One = a['COST (£) charged to Wellcome (inc VAT when charged)']
print ('Plos One stats')
print (Plos_One.describe(),'\n')
b = df[df['Journal title']=='Journal of Biological Chemistry'] 
Journal_BC = b['COST (£) charged to Wellcome (inc VAT when charged)']
print ('Journal of Biological Chemistry stats')
print (Journal_BC.describe(),'\n')

c = df[df['Journal title']=='Proceedings of the National Academy of Sciences'] 
Pnas = c['COST (£) charged to Wellcome (inc VAT when charged)']
print ('Proceedings of the National Academy of Sciences stats')
print (Pnas.describe(),'\n')

d = df[df['Journal title']=='Neuro Image'] 
NI = d['COST (£) charged to Wellcome (inc VAT when charged)']
print('Neuro Image stats')
print (NI.describe(), '\n')

e = df[df['Journal title']=='Nucleic Acids Research'] 
NAR = e['COST (£) charged to Wellcome (inc VAT when charged)']
print('Nucleic Acids Research stats') 
print (NAR.describe())

Plos One stats
count       207.000000
mean      40472.287246
std      193313.084568
min         122.310000
25%         852.480000
50%         899.340000
75%        1044.640000
max      999999.000000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64 

Journal of Biological Chemistry stats
count        64.000000
mean      32666.997031
std      175112.033900
min         265.670000
25%        1158.247500
50%        1332.660000
75%        1716.270000
max      999999.000000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64 

Proceedings of the National Academy of Sciences stats
count        37.000000
mean      27846.971622
std      164260.940970
min         206.320000
25%         625.000000
50%         734.000000
75%         809.620000
max      999999.000000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64 

Neuro Image stats
count      31.000000
mean     2212.181290
std       273.193244
min      1747.160000
25%      20

In [27]:
 journals = df['Journal title']
 df = df.sort_values('Journal title').groupby('Journal title').head(3)

In [29]:
df['Journal title']

21                                     ACSChemicalBiology
20                                     ACSChemicalBiology
19                                     ACSChemicalBiology
22                                ACSChemicalNeuroscience
23                                                ACSNANO
34                                                ACSNano
924                                                 ACTAF
1808                                                  AGE
2115                                                 AIDS
2123                                                 AIDS
2125                                                 AIDS
1797                                            AIDSBehav
1824                                             AIDSCare
1811                                             AIDSCare
2116                                          AIDSJournal
235                                AIDSResearchandTherapy
2118                                               AIDSUK
2117          